clients/da-vinci-client/src/main/java/com/linkedin/davinci/validation/PartitionTracker.java

```diff
@@ -864,22 +891,61 @@ DataValidationException getNewException(Segment segment, DefaultPubSubMessage co
           .append(new Date(producerMetadata.messageTimestamp))
           .append(")");
       if (consumerRecord.getValue().leaderMetadataFooter != null) {
-        sb.append("; leader metadata's upstream offset: ")
+        sb.append("; LeaderMetadata { upstream offset: ")
             .append(consumerRecord.getValue().leaderMetadataFooter.upstreamOffset)
-            .append("; leader metadata's host name: ")
-            .append(consumerRecord.getValue().leaderMetadataFooter.hostName);
+            .append("; upstream pub sub cluster ID: ")
+            .append(consumerRecord.getValue().leaderMetadataFooter.upstreamKafkaClusterId)
+            .append("; producer host name: ")
```

Comment:
Do you think we call it "leader host name" as it comes from `LeaderMetadata.hostName`. We have a similar field called `ProducerMetadata.hostName` in KME, just to avoid any potential ambiguity.

Reply:
I do not see a `hostName` field in the `ProducerMetadata`. Are you sure that is the location where you've seen that?

Regarding the `LeaderMetadata`... there is a bit of history to it... this started off as a piece of data that only the leader server would populate... but at some point, someone changed the behavior so that the `VeniceWriter` would systematically populate it. I don't like it, TBH, as it seems to be just dead weight... but I haven't done the analysis of whether it was ok to strip it off.

In any case, although the object name is now mightily confusing, the truth is that this field now represents the hostname of any producer that wrote into Venice...

Reply:
You are right `ProducerMetadata` doesn't include it and I must have read a wrong file. Also thanks for explaining the history and it makes sense to me now.